### PR TITLE
Fix 'sync-labels' setting

### PR DESCRIPTION
**Pull request by epuertat**
Closes https://github.com/actions/labeler/issues/104 

Following syntax at https://github.com/actions/toolkit/blob/master/packages/glob/README.md#recommended-action-inputs

### CODE REVIEW

**Comment by pje** [comment]
Thanks for the patch, @epuertat! Labeler was behaving incorrectly and your changes corrected that behavior. 

This is technically a breaking change, since labeler will now behave differently with the same inputs out there in the wild.

IMO this necessitates a v4 release and a mention in the changelog of specifically how old yaml configuration will need to be updated to behave "correctly".

Before this change: if you provide literally _any_ value for `sync-labels`, labeler would Sync Labels. (In particular, if you had `sync-labels: false`, labeler would actually Sync Labels. Yikes.)

After this change: you must provide `true` to get Sync Labels behavior. Providing `false` will NOT Sync Labels. (And any other value will be treated like `false`.)

**Comment by epuertat** [comment]
Thanks for taking care of the tests, @pje!

**Comment by tupui** [comment]
Any ETA on merging deploying? We are considering removing this in SciPy because the sync mechanism is annoying us.

**Comment by wknapik** [comment]
Hey. Could we get this merged?

**Comment by hashhar** [comment]
Anything we can do to help move this forward and get this merged?

**Comment by tupui** [comment]
I am sad to say that due to this, we (SciPy) dropped this actions. Hope that this get solved at some point so we can reconsider.

**Comment by jack1142** [comment]
@tupui the workaround for this is fairly easy, you just need to set `sync-labels` to an empty string:
```yaml
sync-labels: ""
```

**Comment by tupui** [comment]
> @tupui the workaround for this is fairly easy, you just need to set `sync-labels` to an empty string:
> 
> ```yaml
> sync-labels: ""
> ```
> 
> Obviously, this should be fixed but no need to drop the action when there's workaround available :P

Oh thanks! We sadly dropped it a few months ago already because there was no action there... I just got notified about this with the recent message and thought I would comment again to say we had to take action–as I feared we might need to.

A "hack" might not make our maintainer team confident I am afraid. I will keep an eye on this.

**Comment by dundargoc** [comment]
bruh, just archive the repository if you've given up on it.

**Comment by jjerphan** [comment]
Hi @pje,

This is a small gentle up: is there anything to be done for this fix to be merged?

Major libraries' CI (like SciPy's and scikit-learn's) would largely benefit from this fix. :pray:

File: src/labeler.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ export async function run() {
   try {
     const token = core.getInput("repo-token", { required: true });
     const configPath = core.getInput("configuration-path", { required: true });
-    const syncLabels = !!core.getInput("sync-labels", { required: false });
+    const syncLabels =
+      core.getInput("sync-labels", { required: false }) === "true";
```
Comment (MaksimZhukov) [inline src/labeler.ts:19]: We can use the `getBooleanInput` core function to parse a boolean input:
```suggestion
    const syncLabels = core.getBooleanInput('sync-labels');
```

**Comment by MaksimZhukov** [comment]
Hello @epuertat!
Thank you for the contribution and sorry for the late review!
I left a comment, please take a look. Also, please rebuild `dist` folder and resolve conflicts.

**Comment by jjerphan** [comment]
@epuertat: Thank you for this contribution, I am looking forward to it being integrated. Are you still working on it? If not, can I supersede it, @MaksimZhukov? :slightly_smiling_face:

**Comment by epuertat** [comment]
@jjerphan @MaksimZhukov @adam-azarchs: please feel free to take over and move forward with this.

**Comment by MaksimZhukov** [comment]
Hello @epuertat ! Thank you for the response!
I am closing the PR in favor of [this one](https://github.com/actions/labeler/pull/480). @jjerphan please find the details in that PR.

**Pull request closed without merging**